roles/openshift_facts/library/openshift_facts.py

```diff
@@ -937,6 +941,16 @@ def set_sdn_facts_if_unset(facts, system_facts):
 
     return facts
 
+def set_nodename(facts):
+    if 'node' in facts and 'common' in facts:
+        if 'cloudprovider' in facts and facts['cloudprovider']['kind'] == 'openstack':
+            facts['node']['nodename'] = facts['provider']['metadata']['hostname'].replace('.novalocal', '')
+        elif 'cloudprovider' in facts and facts['cloudprovider']['kind'] == 'gce':
+            facts['node']['nodename'] = '.'.split(facts['provider']['metadata']['hostname'])[0]
```

Comment:
@detiber The location of the hostname value for gce is found in `facts['provider']['metadata']['instance']['hostname']`.

The following line has proven to be successful:
`facts['node']['nodename'] = facts['provider']['metadata']['instance']['hostname'].split('.')[0]`

This issue is also related to #3172